src/target/riscv/riscv-013.c

```diff
@@ -4568,14 +4608,16 @@ static int write_memory_progbuf(struct target *target, target_addr_t address,
 				DMI_OP_READ, DM_ABSTRACTCS, 0, false, true);
 		if (result != ERROR_OK)
 			goto error;
+
 		while (get_field(abstractcs, DM_ABSTRACTCS_BUSY))
 			if (dm_read(target, &abstractcs, DM_ABSTRACTCS) != ERROR_OK)
 				return ERROR_FAIL;
-		info->cmderr = get_field(abstractcs, DM_ABSTRACTCS_CMDERR);
-		if (info->cmderr == CMDERR_NONE && !dmi_busy_encountered) {
+
+		unsigned int cmderr = get_field(abstractcs, DM_ABSTRACTCS_CMDERR);
+		if (cmderr == CMDERR_NONE && !dmi_busy_encountered) {
 			LOG_TARGET_DEBUG(target, "Successful (partial?) memory write");
-		} else if (info->cmderr == CMDERR_BUSY || dmi_busy_encountered) {
-			if (info->cmderr == CMDERR_BUSY)
+		} else if (cmderr == CMDERR_BUSY || dmi_busy_encountered) {
+			if (cmderr == CMDERR_BUSY)
 				LOG_TARGET_DEBUG(target, "Memory write resulted in abstract command busy response.");
```

Comment:
I would prefer if this change was merged first. https://github.com/riscv/riscv-openocd/pull/959 - is not that big we can wait.

Reply:
ouch. 

> I would prefer if this change was merged first. https://github.com/riscv/riscv-openocd/pull/959 - is not that big we can wait.

it's vice versa. I mean: let's merge https://github.com/riscv/riscv-openocd/pull/959 . This can wait.

Reply:
https://github.com/riscv/riscv-openocd/pull/959 has just been merged. 

So this pull request can now be rebased.